### PR TITLE
Jvm dns ttl

### PR DESCRIPTION
**Pull request by maggarwal13**

### CODE REVIEW

File: whois-commons/src/main/java/net/ripe/db/whois/common/JvmSecurityManager.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@DeployedProfile
+public class JvmSecurityManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JvmSecurityManager.class);
+
+    @Autowired
+    public JvmSecurityManager() {
+        java.security.Security.setProperty("networkaddress.cache.ttl", "30");
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/JvmSecurityManager.java:16]: Can we set this property at runtime? Will it make a difference?

The OpenJDK code suggests this property is only read once (it's in a static block):
https://github.com/AdoptOpenJDK/openjdk-jdk/blob/jdk-11+28/src/java.base/share/classes/sun/net/InetAddressCachePolicy.java#L120

Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/JvmSecurityManager.java:16]: Stack overflow to the rescue:

`This code must execute before any other code in the JVM attempts to perform networking operations.`

Is it enough to also put these properties in a static block?

https://github.com/AdoptOpenJDK/openjdk-jdk/blob/jdk-11+28/src/java.base/share/classes/sun/net/InetAddressCachePolicy.java#L120

Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/JvmSecurityManager.java:16]: A workaround to not setting the security properties is to set the *system* properties `sun.net.inetaddr.ttl` and `sun.net.inetaddr.negative.ttl`. They are deprecated but work for now.

https://stackoverflow.com/questions/41708885/best-way-to-set-networkaddress-cache-ttl

Comment (maggarwal13) [reply whois-commons/src/main/java/net/ripe/db/whois/common/JvmSecurityManager.java:16]: There is another way to provide security file but let's have a discussion about it, we need to think about implications of it.
https://dzone.com/articles/how-override-java-security

Comment (maggarwal13) [reply whois-commons/src/main/java/net/ripe/db/whois/common/JvmSecurityManager.java:16]: Lets try not to use the deprecated methods first if we can find better way to do this

**Pull request closed without merging**